src/transformers/models/whisper/configuration_whisper.py

```diff
@@ -214,3 +222,115 @@ def __init__(
             begin_suppress_tokens=begin_suppress_tokens,
             **kwargs,
         )
+
+
+class WhisperEncoderOnnxConfig(OnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("input_features", {0: "batch", 1: "feature_size", 2: "encoder_sequence"}),
+            ]
+        )
+
+    @property
+    def outputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict({"last_hidden_state": {0: "batch", 1: "encoder_sequence"}})
+
+
+class WhisperDecoderOnnxConfig(OnnxSeq2SeqConfigWithPast):
```

Comment:
This config handling the export of encoder/decoder model have already been integrated to [optimum](https://github.com/huggingface/optimum/blob/v1.4.0/optimum/onnx/configuration.py#L282). Given the current efforts to migrate the ONNX export from transformers to optimum, I don't think it makes sense to duplicate this classes in transformers (given that it will be soon depreciated). Why not move this PR to optimum ? @mht-sharma  @lewtun @michaelbenayoun

Reply:
Good point @echarlaix - I originally thought this PR was needed to enable the `from_transformers=True` feature in `optimum`. I now agree this would be best moved to `optimum` as part of https://github.com/huggingface/optimum/pull/420

WDYT @mht-sharma? Shall we close this PR and move the logic over?

Reply:
Hi @lewtun @echarlaix we may not be able to close this PR completely. Optimum still uses some of the components from the transformers.onnx package. Hence, to support the above model we would need to create placeholders for WhisperOnnxConfig, support for speech modality, AutoClassForSpeechSeq2Seq, audio data generation etc in the transformers repository.

For example
https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/modeling_seq2seq.py#L433

```python
        model = FeaturesManager.get_model_from_feature(cls.export_feature, model_id)
        _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=cls.export_feature)
        onnx_config = model_onnx_config(model.config)
```
The above line requires that the model is supported in transformers.onnx also.

Best we could do is to copy the Whisper ONNX configs in optimum also and use for optimum pipelines.

Reply:
Thanks for clarifying the need for this PR @mht-sharma - I overlooked this dependency in `optimum` 🙈

In the interest of moving fast + bringing value to the community, it probably makes sense to indeed continue with this PR in `transformers` since the ONNX migration to `optimum` will probably take some more time. WDYT @echarlaix ?

Reply:
We need an `WhisperOnnxConfig` such as [`T5OnnxConfig`](https://github.com/huggingface/transformers/blob/v4.23.1/src/transformers/models/t5/configuration_t5.py#L143) which would keep the export behavior of whisper similar to other encoder/decoder model of `transformers.onnx` (export of entire model in one shot) but keep the `EncoderOnnxConfig` / `DecoderOnnxConfig` in `optimum`

Reply:
What do you think @mht-sharma @lewtun ?

Reply:
OK, if we can enable Whisper to run on ONNX Runtime in `optimum` without any dependency on internal `transformers` logic, then I think @echarlaix's proposal to migrate everything there will help us be future proof. It will also make the final PR a bit easier to test :)

Apologies if I caused confusion by flipping back and forth - this will become much simpler once we have the ONNX exporter in `optimum`!

Reply:
Thanks @lewtun @echarlaix  I will migrate the changes.